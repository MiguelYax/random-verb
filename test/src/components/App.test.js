import React from 'react';
import renderer from 'react-test-renderer';

let App;
jest.mock('../../../src/components/Random', () => ({
    getRandomVerb: () => ({
        "verb": "Wake up",
        "past": "Woke up",
        "pastParticiple": "Woken up",
        "meaning": "Despertar",
        "finalSound": ""
    })
}));

describe('when component src/components/App.jsx is used', () => {
    beforeEach(() => {
        App = require('../../../src/components/App').default;
    })
    it('shold be render main components', () => {
        const app = renderer.create(<App />).toJSON();
        expect(app).toMatchSnapshot();
    });
});
