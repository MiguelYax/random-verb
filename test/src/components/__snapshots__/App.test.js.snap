// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when component src/components/App.jsx is used shold be render main components 1`] = `
Array [
  <nav
    className="navbar navbar-expand-xl navbar-light bg-light text-primary"
  >
    <a
      className="text-primary navbar-brand"
      href="#home"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-random fa-w-16 mr-2"
        data-icon="random"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504.971 359.029c9.373 9.373 9.373 24.569 0 33.941l-80 79.984c-15.01 15.01-40.971 4.49-40.971-16.971V416h-58.785a12.004 12.004 0 0 1-8.773-3.812l-70.556-75.596 53.333-57.143L352 336h32v-39.981c0-21.438 25.943-31.998 40.971-16.971l80 79.981zM12 176h84l52.781 56.551 53.333-57.143-70.556-75.596A11.999 11.999 0 0 0 122.785 96H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12zm372 0v39.984c0 21.46 25.961 31.98 40.971 16.971l80-79.984c9.373-9.373 9.373-24.569 0-33.941l-80-79.981C409.943 24.021 384 34.582 384 56.019V96h-58.785a12.004 12.004 0 0 0-8.773 3.812L96 336H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h110.785c3.326 0 6.503-1.381 8.773-3.812L352 176h32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Random Verb
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <form
        className="form-inline"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          id="darkModeButton"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-adjust fa-w-16 "
            data-icon="adjust"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </nav>,
  <div
    className="vh-100 mw-100 container"
  >
    <div
      className="h-75 pt-5 justify-content-md-center row"
    >
      <div
        className="col-lg-10 col-md-10"
      >
        <table
          className="text-primary table table-light table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th
                colSpan="2"
              >
                Despertar
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Verb
              </td>
              <td>
                Wake up
              </td>
            </tr>
            <tr>
              <td>
                Past
              </td>
              <td>
                Woke up
              </td>
            </tr>
            <tr>
              <td>
                Past participle
              </td>
              <td>
                Woken up
              </td>
            </tr>
            
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="pt-2 justify-content-center row row-cols-lg-10 row-cols-md-10"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
         Randomize
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-sync-alt fa-w-16 ml-2"
          data-icon="sync-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
    <div
      className=" pt-3 justify-content-center row"
    >
      <div
        className="text-primary"
      >
        Version 
        0.0.1
         
      </div>
    </div>
  </div>,
]
`;
